### PR TITLE
Update beats - switch from nosetest to pytest

### PR DESCRIPTION
**Pull request by simitt**
## Motivation/summary
Switch from `nosetest`  to `pytest` for python system tests following the changes in libbeat https://github.com/elastic/beats/pull/16883/files.

The PR fixes some deprecation warnings, but since we are moving to go systemtests I did not bother to address all (some are coming from beats). 

## Checklist

~- [ ] I have signed the [Contributor License Agreement](https://www.elastic.co/contributor-agreement/).~
~- [ ] I have updated [CHANGELOG.asciidoc](https://github.com/elastic/apm-server/blob/master/CHANGELOG.asciidoc)~

I have considered changes for:
~- [ ] documentation~
~- [ ] logging (add log lines, choose appropriate log selector, etc.)~
~- [ ] metrics and monitoring (create issue for Kibana team to add metrics to visualizations, e.g. [Kibana#44001](https://github.com/elastic/kibana/issues/44001))~
- [ ] automated tests (add tests for the code changes, all [**unit** tests](https://github.com/elastic/apm-server/blob/master/TESTING.md) pass locally)
~- [ ] telemetry~
~- [ ] Elasticsearch Service (https://cloud.elastic.co)~
~- [ ] Elastic Cloud Enterprise (https://www.elastic.co/products/ece)~
~- [ ] Elastic Cloud on Kubernetes (https://www.elastic.co/elastic-cloud-kubernetes)~

## How to test these changes
run `INTEGRATION_TESTS=1 make system-tests` 

## Related issues
closes https://github.com/elastic/apm-server/issues/4059

### CODE REVIEW

File: tests/system/apmserver.py [hunk 1]
```diff
@@ -30,6 +29,10 @@
 diagnostic_interval = float(os.environ.get('DIAGNOSTIC_INTERVAL', 0))
 
 
+def is_subset(s, d):
+    return all(d.get(key, None) == val for key, val in s.items())
```
Comment (axw) [inline tests/system/apmserver.py:33]: Can we use [set.issubset](https://docs.python.org/3.7/library/stdtypes.html#frozenset.issubset) from stdlib?

File: tests/system/test_export.py [hunk 2]
```diff
@@ -27,7 +28,7 @@ def test_export_config(self):
         """
         Test export default config
         """
-        config = yaml.load(self.command_output)
+        config = yaml.load(self.command_output, Loader=Loader)
```
Comment (axw) [inline tests/system/test_export.py:31]: What caused this change? Did the PyYAML version get updated?

File: tests/system/test_integration_acm.py [hunk 3]
```diff
@@ -162,12 +162,12 @@ def test_config_requests(self):
...
             "response_code": 200,
         })
-        self.assertDictEqual({"transaction_sample_rate": "0.05"}, r5.json())
+        self.assertEqual({"transaction_sample_rate": "0.05"}, r5.json())
 
         config_request_logs = list(self.logged_requests(url="/config/v1/agents"))
         assert len(config_request_logs) == len(expect_log)
         for want, got in zip(expect_log, config_request_logs):
-            self.assertDictContainsSubset(want, got)
+            assert is_subset(want, got)
```
Comment (axw) [inline tests/system/test_integration_acm.py:170]:
```suggestion
            assert set(want).issubset(got)
```

?

File: tests/system/test_export.py (hunk 2 above)
Comment (simitt) [reply tests/system/test_export.py:31]: yes it changed from `4.2` to `5.3`, which led to a warning when not specifying the `Loader`.

File: tests/system/apmserver.py (hunk 1 above)
Comment (simitt) [reply tests/system/apmserver.py:33]: yes; didn't bother to convert to `set` just for the check (shouldn't make a difference performance wise) but agree it's better to use the existing one.

**Pull request merged**